cli/filter_data_through_cmd.go

```diff
@@ -0,0 +1,29 @@
+package cli
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/google/shlex"
+	"os/exec"
+	"strings"
+)
+
+func filterDataThroughCmd(data []byte, filter string, subject, stream string) ([]byte, error) {
+	parts, err := shlex.Split(filter)
+	if err != nil {
+		return nil, fmt.Errorf("the filter command line could not be parsed: %w", err)
+	}
+	cmd := parts[0]
+	args := parts[1:]
+	// maybe we want to replace some strings to give the filter something to reason about?
+	for idx := range args {
+		arg := &args[idx]
+		*arg = strings.ReplaceAll(*arg, "{{subject}}", subject)
+		*arg = strings.ReplaceAll(*arg, "{{stream}}", stream)
```

Comment:
elsewhere we use `text/template` for this and people would already be used to being able to put spaces in and so forth.  So lets also use `text/template` here and then make it `Subject` and `Stream` also for consistency with like whats in use in `nats req` and so forth wrt capitalization

Reply:
Using text/template also means we can easily extend later

Reply:
I need to read up in text/template. I never used that myself.

Reply:
I changed it to use the `text/template` package. Please check. I wonder about how there is sometimes `{{.Count}}` (README.me:145) but mostly `{{Count}}` in the code. Am I supposed to use the functions instead of the fields or doing both, what I think that is being done in `pubReplyBodyTemplate()`?

Reply:
I think the readme is probably wrong. Can pass in a map[string]any wirh those fields is enough and then highlighting them in help output.

Reply:
Do you mean additionally to the `{{.Subject}}` or instead? I ask because `pubReplyBodyTemplate()` allows both types.

Reply:
you can pick one, which ever you like - the pub/reply one is complex because of a early backward incompatible change and so i made it kind of work.  since yours is new you can just go with one as long as its in --help

Reply:
Well. I picked the `{{.Subject}}` type and updated the README with the last commit already. The reason was that using `{{.Subject}}` instead of `{{Subject}}` may people aware of that it is using the text/template and let them use other functionality of it. But I got unsure because most of the documentation does not use this variant and the additional `.` may make people wonder. So, the `{{Subject}}` variant it will be then. The documentation was updated.

Reply:
looks good thanks, any chance we can get that duplicated code pulled into a seperate function?

Reply:
I am unsure what duplicate code you refer to.